tool/mapeditor.re

```diff
@@ -72,9 +90,40 @@ fn create_map_editor_menu() -> Ui {
                         } else {
                             MAP_EDITOR_STATE.map = Tas::current_map();
                         };
-                        MAP_EDITOR_LABEL.text = "Stop Map Editor";
+                        MAP_EDITOR_EDIT_LABEL.text = "Stop editing map";
                         add_component(MAP_EDITOR_COMPONENT);
                         add_component(MOVEMENT_COMPONENT);
+                        MAP_EDITOR_STATE.mode = MapMode::Edit;
+                        MOVEMENT_STATE.enable_fly = false;
+                        leave_ui();
+                        leave_ui();
+                        leave_ui();
+                    }));
+                }
+            },
+        }),
+        UiElement::Button(UiButton {
```

Comment:
1. When the map editor is running, there should only be a single button to stop it.
2. These two buttons share most of their code, which could be refactored out. (Keep in mind that we are inside a plain old function which returns the UI Elements.)